### PR TITLE
Integration RPC tests

### PR DESCRIPTION
**Pull request by niklabh**
**Pull Request Summary**

Added mocha test suite for rpc node. This PR aims to introduce a framework for testing rpc node with typescript tests. This will help us test collator end to end.

Steps:

Download astar collator binary from releases
Spawn a local test network using child_process
Run test suite against that that test network

**Check list**
- [ ] contains breaking changes
- [x] adds new feature
- [ ] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

**This pull request makes the following changes:**

**Adds**
- test suite

### CODE REVIEW

**Review by impelcrypto — APPROVED** [review]

**Review by sirius651 — APPROVED** [review]

File: rpc-tests/tests/test-rpc.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        const keyring = new Keyring({ type: 'sr25519' });
+      
+        // Add Alice to our keyring with a hard-derivation path (empty phrase, so uses dev)
+        const alice = keyring.addFromUri('//Alice');
+
+        // Create a extrinsic, transferring 100 units to Bob
+        const transfer = api.tx.dappsStaking.register(getAddressEnum(CONTRACT));
+      
+        // Sign and send the transaction using our account
+        const hash = await transfer.signAndSend(alice, { nonce: -1 });
```
Comment (bobo-k2) [inline rpc-tests/tests/test-rpc.js:43]: As we discussed a few days ago. You will get hash as soon as node accepts transaction, which doesn't necessarily means that transaction will pass. Transaction events need to be parsed in order to determine if transaction succeed or not.  ref. https://github.com/AstarNetwork/astar-apps/blob/9fc092904f056a56f9b76a67f481b12b20cf0b39/src/modules/extrinsic/utils/index.ts#L26

**Review by bobo-k2 — CHANGES_REQUESTED** [review]

**Comment by bobo-k2** [comment]
@niklabh please update this PR after you merge https://github.com/AstarNetwork/Astar/pull/709

**Comment by niklabh** [comment]
Updated

**Review by bobo-k2 — APPROVED** [review]
LGTM

**Pull request merged**